Formula/docker-buildx.rb

```diff
@@ -0,0 +1,37 @@
+class DockerBuildx < Formula
+  desc "Docker CLI plugin for extended build capabilities with BuildKit"
+  homepage "https://docs.docker.com/buildx/working-with-buildx/"
+  url "https://github.com/docker/buildx.git",
+      tag:      "v0.8.2",
+      revision: "6224def4dd2c3d347eee19db595348c50d7cb491"
+  license "Apache-2.0"
+  head "https://github.com/docker/buildx.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/docker/buildx/version.Version=v#{version}
+      -X github.com/docker/buildx/version.Revision=#{Utils.git_head}
+    ]
+
+    system "go", "build", *std_go_args(ldflags: ldflags), "./cmd/buildx"
+
+    doc.install Dir["docs/reference/*.md"]
+  end
+
+  def caveats
+    <<~EOS
+      docker-buildx is a Docker plugin. For Docker to find this plugin, symlink it:
+        mkdir -p ~/.docker/cli-plugins
+        ln -sfn #{opt_bin}/docker-buildx ~/.docker/cli-plugins/docker-buildx
+    EOS
+  end
+
+  test do
+    assert_match "github.com/docker/buildx v#{version}", shell_output("#{bin}/docker-buildx version")
+    output = shell_output(bin/"docker-buildx build . 2>&1", 1)
+    assert_match "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?", output
```

Comment:
I think we'll need to account for a different error on Linux, so this should work:
```suggestion
    assert_match(/(denied while trying to|Cannot) connect to the Docker daemon/, output)
```